ahrefs/monorobot
# Add tests for multi-repo setup

## Description of the task

(depends on #97)

- Refactor test suite so it can load secrets and config files on a per-test-case basis.
- Add tests to confirm behavior related to the multi-repo setup additions from #97 

## How to test

Existing tests should pass without regression. Three additional test cases confirm additional behavior related to this multi-repo setup. Namely:

- If the `allowed_repos` list excludes the payload repo, it should be ignored. (`status.multi_repo_disallowed_repo`)
- The status rules should be evaluated for the correct repo. In this test case, it should correctly detect that `master`'s last build status was a failure for the given payload repo, and issue a success notification.(`status.multi_repo_independent_status_state`)

```
make test
```

## References

- multi-repo setup PR: #97
